// @backstage/canon/dist/components/ScrollArea/ScrollArea.esm.js
import React__default from 'react';
import { ScrollArea as ScrollArea$1 } from '@base-ui-components/react/scroll-area';
import clsx from 'clsx';

const ScrollAreaRoot = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  ScrollArea$1.Root,
  {
    ref,
    className: clsx("canon-ScrollAreaRoot", className),
    ...props
  }
));
ScrollAreaRoot.displayName = ScrollArea$1.Root.displayName;
const ScrollAreaViewport = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  ScrollArea$1.Viewport,
  {
    ref,
    className: clsx("canon-ScrollAreaViewport", className),
    ...props
  }
));
ScrollAreaViewport.displayName = ScrollArea$1.Viewport.displayName;
const ScrollAreaScrollbar = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  ScrollArea$1.Scrollbar,
  {
    ref,
    className: clsx("canon-ScrollAreaScrollbar", className),
    ...props
  }
));
ScrollAreaScrollbar.displayName = ScrollArea$1.Scrollbar.displayName;
const ScrollAreaThumb = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  ScrollArea$1.Thumb,
  {
    ref,
    className: clsx("canon-ScrollAreaThumb", className),
    ...props
  }
));
ScrollAreaThumb.displayName = ScrollArea$1.Thumb.displayName;
const ScrollArea = {
  Root: ScrollAreaRoot,
  Viewport: ScrollAreaViewport,
  Scrollbar: ScrollAreaScrollbar,
  Thumb: ScrollAreaThumb
};

export { ScrollArea };
