// @backstage/plugin-scaffolder-node-test-utils/dist/actions/mockActionContext.cjs.js
'use strict';

var stream = require('stream');
var backendTestUtils = require('@backstage/backend-test-utils');
var loggerToWinstonLogger = require('./loggerToWinstonLogger.cjs.js');

const createMockActionContext = (options) => {
  const credentials = backendTestUtils.mockCredentials.user();
  const defaultContext = {
    logger: loggerToWinstonLogger.loggerToWinstonLogger(backendTestUtils.mockServices.logger.mock()),
    logStream: new stream.PassThrough(),
    output: jest.fn(),
    createTemporaryDirectory: jest.fn(),
    input: {},
    async checkpoint(opts) {
      return opts.fn();
    },
    getInitiatorCredentials: () => Promise.resolve(credentials),
    task: {
      id: "mock-task-id"
    }
  };
  const createDefaultWorkspace = () => ({
    workspacePath: backendTestUtils.createMockDirectory().resolve("workspace")
  });
  if (!options) {
    return {
      ...defaultContext,
      ...createDefaultWorkspace()
    };
  }
  const {
    input,
    logger,
    logStream,
    secrets,
    templateInfo,
    workspacePath,
    task
  } = options;
  return {
    ...defaultContext,
    ...workspacePath ? { workspacePath } : createDefaultWorkspace(),
    ...workspacePath && {
      createTemporaryDirectory: jest.fn().mockResolvedValue(workspacePath)
    },
    ...logger && { logger },
    ...logStream && { logStream },
    ...input && { input },
    ...secrets && { secrets },
    ...task && { task },
    templateInfo
  };
};

exports.createMockActionContext = createMockActionContext;
