// @backstage/plugin-catalog/dist/alpha/pages.esm.js
import React from 'react';
import { convertLegacyRouteRef, compatWrapper } from '@backstage/core-compat-api';
import { PageBlueprint, createExtensionInput, coreExtensionData } from '@backstage/frontend-plugin-api';
import { entityRouteRef, AsyncEntityProvider } from '@backstage/plugin-catalog-react';
import { EntityHeaderBlueprint, EntityContentBlueprint, defaultEntityContentGroups } from '@backstage/plugin-catalog-react/alpha';
import { rootRouteRef } from '../routes.esm.js';
import { useEntityFromUrl } from '../components/CatalogEntityPage/useEntityFromUrl.esm.js';
import { buildFilterFn } from './filter/FilterWrapper.esm.js';
import { EntityHeader } from './components/EntityHeader/EntityHeader.esm.js';

const catalogPage = PageBlueprint.makeWithOverrides({
  inputs: {
    filters: createExtensionInput([coreExtensionData.reactElement])
  },
  factory(originalFactory, { inputs }) {
    return originalFactory({
      defaultPath: "/catalog",
      routeRef: convertLegacyRouteRef(rootRouteRef),
      loader: async () => {
        const { BaseCatalogPage } = await import('../components/CatalogPage/index.esm.js');
        const filters = inputs.filters.map(
          (filter) => filter.get(coreExtensionData.reactElement)
        );
        return compatWrapper(/* @__PURE__ */ React.createElement(BaseCatalogPage, { filters: /* @__PURE__ */ React.createElement(React.Fragment, null, filters) }));
      }
    });
  }
});
const catalogEntityPage = PageBlueprint.makeWithOverrides({
  name: "entity",
  inputs: {
    header: createExtensionInput(
      [EntityHeaderBlueprint.dataRefs.element.optional()],
      { singleton: true, optional: true }
    ),
    contents: createExtensionInput([
      coreExtensionData.reactElement,
      coreExtensionData.routePath,
      coreExtensionData.routeRef.optional(),
      EntityContentBlueprint.dataRefs.title,
      EntityContentBlueprint.dataRefs.filterFunction.optional(),
      EntityContentBlueprint.dataRefs.filterExpression.optional(),
      EntityContentBlueprint.dataRefs.group.optional()
    ])
  },
  config: {
    schema: {
      groups: (z) => z.array(z.record(z.string(), z.object({ title: z.string() }))).optional()
    }
  },
  factory(originalFactory, { config, inputs }) {
    return originalFactory({
      defaultPath: "/catalog/:namespace/:kind/:name",
      routeRef: convertLegacyRouteRef(entityRouteRef),
      loader: async () => {
        const { EntityLayout } = await import('./components/EntityLayout/index.esm.js');
        const header = inputs.header?.get(
          EntityHeaderBlueprint.dataRefs.element
        ) ?? /* @__PURE__ */ React.createElement(EntityHeader, null);
        let groups = Object.entries(defaultEntityContentGroups).reduce(
          (rest, group) => {
            const [groupId, groupValue] = group;
            return {
              ...rest,
              [groupId]: { title: groupValue, items: [] }
            };
          },
          {}
        );
        if (config.groups) {
          groups = config.groups.reduce((rest, group) => {
            const [groupId, groupValue] = Object.entries(group)[0];
            return {
              ...rest,
              [groupId]: { title: groupValue.title, items: [] }
            };
          }, {});
        }
        for (const output of inputs.contents) {
          const itemId = output.node.spec.id;
          const itemTitle = output.get(EntityContentBlueprint.dataRefs.title);
          const itemGroup = output.get(EntityContentBlueprint.dataRefs.group);
          const group = itemGroup && groups[itemGroup];
          if (!group) {
            groups[itemId] = { title: itemTitle, items: [output] };
            continue;
          }
          group.items.push(output);
        }
        const Component = () => {
          return /* @__PURE__ */ React.createElement(AsyncEntityProvider, { ...useEntityFromUrl() }, /* @__PURE__ */ React.createElement(EntityLayout, { header }, Object.values(groups).flatMap(
            ({ title, items }) => items.map((output) => /* @__PURE__ */ React.createElement(
              EntityLayout.Route,
              {
                group: title,
                key: output.get(coreExtensionData.routePath),
                path: output.get(coreExtensionData.routePath),
                title: output.get(EntityContentBlueprint.dataRefs.title),
                if: buildFilterFn(
                  output.get(
                    EntityContentBlueprint.dataRefs.filterFunction
                  ),
                  output.get(
                    EntityContentBlueprint.dataRefs.filterExpression
                  )
                )
              },
              output.get(coreExtensionData.reactElement)
            ))
          )));
        };
        return compatWrapper(/* @__PURE__ */ React.createElement(Component, null));
      }
    });
  }
});
var pages = [catalogPage, catalogEntityPage];

export { catalogEntityPage, catalogPage, pages as default };
