// @backstage/canon/dist/components/Menu/Menu.esm.js
import React__default from 'react';
import { Menu as Menu$1 } from '@base-ui-components/react/menu';
import clsx from 'clsx';

const MenuTrigger = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Trigger,
  {
    ref,
    className: clsx("canon-MenuTrigger", className),
    ...props
  }
));
MenuTrigger.displayName = Menu$1.Trigger.displayName;
const MenuBackdrop = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Backdrop,
  {
    ref,
    className: clsx("canon-MenuBackdrop", className),
    ...props
  }
));
MenuBackdrop.displayName = Menu$1.Backdrop.displayName;
const MenuPositioner = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Positioner,
  {
    ref,
    className: clsx("canon-MenuPositioner", className),
    ...props
  }
));
MenuPositioner.displayName = Menu$1.Positioner.displayName;
const MenuPopup = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Popup,
  {
    ref,
    className: clsx("canon-MenuPopup", className),
    ...props
  }
));
MenuPopup.displayName = Menu$1.Popup.displayName;
const MenuArrow = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Arrow,
  {
    ref,
    className: clsx("canon-MenuArrow", className),
    ...props
  }
));
MenuArrow.displayName = Menu$1.Arrow.displayName;
const MenuItem = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Item,
  {
    ref,
    className: clsx("canon-MenuItem", className),
    ...props
  }
));
MenuItem.displayName = Menu$1.Item.displayName;
const MenuGroup = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Group,
  {
    ref,
    className: clsx("canon-MenuGroup", className),
    ...props
  }
));
MenuGroup.displayName = Menu$1.Group.displayName;
const MenuGroupLabel = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.GroupLabel,
  {
    ref,
    className: clsx("canon-MenuGroupLabel", className),
    ...props
  }
));
MenuGroupLabel.displayName = Menu$1.GroupLabel.displayName;
const MenuRadioGroup = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.RadioGroup,
  {
    ref,
    className: clsx("canon-MenuRadioGroup", className),
    ...props
  }
));
MenuRadioGroup.displayName = Menu$1.RadioGroup.displayName;
const MenuRadioItem = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.RadioItem,
  {
    ref,
    className: clsx("canon-MenuRadioItem", className),
    ...props
  }
));
MenuRadioItem.displayName = Menu$1.RadioItem.displayName;
const MenuRadioItemIndicator = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.RadioItemIndicator,
  {
    ref,
    className: clsx("canon-MenuRadioItemIndicator", className),
    ...props
  }
));
MenuRadioItemIndicator.displayName = Menu$1.RadioItemIndicator.displayName;
const MenuCheckboxItem = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.CheckboxItem,
  {
    ref,
    className: clsx("canon-MenuCheckboxItem", className),
    ...props
  }
));
MenuCheckboxItem.displayName = Menu$1.CheckboxItem.displayName;
const MenuCheckboxItemIndicator = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.CheckboxItemIndicator,
  {
    ref,
    className: clsx("canon-MenuCheckboxItemIndicator", className),
    ...props
  }
));
MenuCheckboxItemIndicator.displayName = Menu$1.CheckboxItemIndicator.displayName;
const MenuSubmenuTrigger = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.SubmenuTrigger,
  {
    ref,
    className: clsx("canon-MenuSubmenuTrigger", className),
    ...props
  }
));
MenuSubmenuTrigger.displayName = Menu$1.SubmenuTrigger.displayName;
const MenuSeparator = React__default.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React__default.createElement(
  Menu$1.Separator,
  {
    ref,
    className: clsx("canon-MenuSeparator", className),
    ...props
  }
));
MenuSeparator.displayName = Menu$1.Separator.displayName;
const Menu = {
  Root: Menu$1.Root,
  Trigger: MenuTrigger,
  Portal: Menu$1.Portal,
  Backdrop: MenuBackdrop,
  Positioner: MenuPositioner,
  Popup: MenuPopup,
  Arrow: MenuArrow,
  Item: MenuItem,
  Group: MenuGroup,
  GroupLabel: MenuGroupLabel,
  RadioGroup: MenuRadioGroup,
  RadioItem: MenuRadioItem,
  RadioItemIndicator: MenuRadioItemIndicator,
  CheckboxItem: MenuCheckboxItem,
  CheckboxItemIndicator: MenuCheckboxItemIndicator,
  SubmenuTrigger: MenuSubmenuTrigger,
  Separator: MenuSeparator
};

export { Menu };
