// @backstage/backend-defaults/dist/entrypoints/permissionsRegistry/permissionsRegistryServiceFactory.cjs.js
'use strict';

var backendPluginApi = require('@backstage/backend-plugin-api');
var pluginPermissionNode = require('@backstage/plugin-permission-node');

function assertRefPluginId(ref, pluginId) {
  if (ref.pluginId !== pluginId) {
    throw new Error(
      `Resource type '${ref.resourceType}' belongs to plugin '${ref.pluginId}', but was used with plugin '${pluginId}'`
    );
  }
}
const permissionsRegistryServiceFactory = backendPluginApi.createServiceFactory({
  service: backendPluginApi.coreServices.permissionsRegistry,
  deps: {
    lifecycle: backendPluginApi.coreServices.lifecycle,
    httpRouter: backendPluginApi.coreServices.httpRouter,
    pluginMetadata: backendPluginApi.coreServices.pluginMetadata
  },
  async factory({ httpRouter, lifecycle, pluginMetadata }) {
    const router = pluginPermissionNode.createPermissionIntegrationRouter();
    const pluginId = pluginMetadata.getId();
    httpRouter.use(router);
    let started = false;
    lifecycle.addStartupHook(() => {
      started = true;
    });
    return {
      addResourceType(resource) {
        if (started) {
          throw new Error(
            "Cannot add permission resource types after the plugin has started"
          );
        }
        assertRefPluginId(resource.resourceRef, pluginId);
        router.addResourceType({
          ...resource,
          resourceType: resource.resourceRef.resourceType
        });
      },
      addPermissions(permissions) {
        if (started) {
          throw new Error(
            "Cannot add permissions after the plugin has started"
          );
        }
        router.addPermissions(permissions);
      },
      addPermissionRules(rules) {
        if (started) {
          throw new Error(
            "Cannot add permission rules after the plugin has started"
          );
        }
        router.addPermissionRules(rules);
      },
      getPermissionRuleset(resourceRef) {
        assertRefPluginId(resourceRef, pluginId);
        return router.getPermissionRuleset(resourceRef);
      }
    };
  }
});

exports.permissionsRegistryServiceFactory = permissionsRegistryServiceFactory;
