// @backstage/plugin-scaffolder-backend/dist/scaffolder/actions/deprecated.cjs.js
'use strict';

var pluginScaffolderBackendModuleGithub = require('@backstage/plugin-scaffolder-backend-module-github');
var pluginScaffolderBackendModuleGitlab = require('@backstage/plugin-scaffolder-backend-module-gitlab');
var pluginScaffolderBackendModuleAzure = require('@backstage/plugin-scaffolder-backend-module-azure');
var pluginScaffolderBackendModuleBitbucket = require('@backstage/plugin-scaffolder-backend-module-bitbucket');
var pluginScaffolderBackendModuleBitbucketCloud = require('@backstage/plugin-scaffolder-backend-module-bitbucket-cloud');
var pluginScaffolderBackendModuleBitbucketServer = require('@backstage/plugin-scaffolder-backend-module-bitbucket-server');
var pluginScaffolderBackendModuleGerrit = require('@backstage/plugin-scaffolder-backend-module-gerrit');

const createGithubActionsDispatchAction = pluginScaffolderBackendModuleGithub.createGithubActionsDispatchAction;
const createGithubDeployKeyAction = pluginScaffolderBackendModuleGithub.createGithubDeployKeyAction;
const createGithubEnvironmentAction = pluginScaffolderBackendModuleGithub.createGithubEnvironmentAction;
const createGithubIssuesLabelAction = pluginScaffolderBackendModuleGithub.createGithubIssuesLabelAction;
const createGithubRepoCreateAction = pluginScaffolderBackendModuleGithub.createGithubRepoCreateAction;
const createGithubRepoPushAction = pluginScaffolderBackendModuleGithub.createGithubRepoPushAction;
const createGithubWebhookAction = pluginScaffolderBackendModuleGithub.createGithubWebhookAction;
const createPublishGithubAction = pluginScaffolderBackendModuleGithub.createPublishGithubAction;
const createPublishGithubPullRequestAction = pluginScaffolderBackendModuleGithub.createPublishGithubPullRequestAction;
const createPublishBitbucketAction = pluginScaffolderBackendModuleBitbucket.createPublishBitbucketAction;
const createPublishBitbucketCloudAction = pluginScaffolderBackendModuleBitbucketCloud.createPublishBitbucketCloudAction;
const createPublishBitbucketServerAction = pluginScaffolderBackendModuleBitbucketServer.createPublishBitbucketServerAction;
const createPublishBitbucketServerPullRequestAction = pluginScaffolderBackendModuleBitbucketServer.createPublishBitbucketServerPullRequestAction;
const createPublishAzureAction = pluginScaffolderBackendModuleAzure.createPublishAzureAction;
const createPublishGerritAction = pluginScaffolderBackendModuleGerrit.createPublishGerritAction;
const createPublishGerritReviewAction = pluginScaffolderBackendModuleGerrit.createPublishGerritReviewAction;
const createPublishGitlabAction = pluginScaffolderBackendModuleGitlab.createPublishGitlabAction;
const createPublishGitlabMergeRequestAction = pluginScaffolderBackendModuleGitlab.createPublishGitlabMergeRequestAction;

exports.createGithubActionsDispatchAction = createGithubActionsDispatchAction;
exports.createGithubDeployKeyAction = createGithubDeployKeyAction;
exports.createGithubEnvironmentAction = createGithubEnvironmentAction;
exports.createGithubIssuesLabelAction = createGithubIssuesLabelAction;
exports.createGithubRepoCreateAction = createGithubRepoCreateAction;
exports.createGithubRepoPushAction = createGithubRepoPushAction;
exports.createGithubWebhookAction = createGithubWebhookAction;
exports.createPublishAzureAction = createPublishAzureAction;
exports.createPublishBitbucketAction = createPublishBitbucketAction;
exports.createPublishBitbucketCloudAction = createPublishBitbucketCloudAction;
exports.createPublishBitbucketServerAction = createPublishBitbucketServerAction;
exports.createPublishBitbucketServerPullRequestAction = createPublishBitbucketServerPullRequestAction;
exports.createPublishGerritAction = createPublishGerritAction;
exports.createPublishGerritReviewAction = createPublishGerritReviewAction;
exports.createPublishGithubAction = createPublishGithubAction;
exports.createPublishGithubPullRequestAction = createPublishGithubPullRequestAction;
exports.createPublishGitlabAction = createPublishGitlabAction;
exports.createPublishGitlabMergeRequestAction = createPublishGitlabMergeRequestAction;
