// @backstage/canon/dist/components/Table/Table.esm.js
import * as React from 'react';

const Table = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement("div", { className: "table" }, /* @__PURE__ */ React.createElement("table", { ref, className, ...props })));
Table.displayName = "Table";
const TableHeader = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement(
  "thead",
  {
    ref,
    className: ["table-header", className].join(" "),
    ...props
  }
));
TableHeader.displayName = "TableHeader";
const TableBody = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement("tbody", { ref, className: ["table-body", className].join(" "), ...props }));
TableBody.displayName = "TableBody";
const TableFooter = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement(
  "tfoot",
  {
    ref,
    className: ["table-footer", className].join(" "),
    ...props
  }
));
TableFooter.displayName = "TableFooter";
const TableRow = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement("tr", { ref, className: ["table-row", className].join(" "), ...props }, props.children));
TableRow.displayName = "TableRow";
const TableHead = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement("th", { ref, className: ["table-head", className].join(" "), ...props }));
TableHead.displayName = "TableHead";
const TableCell = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement("td", { ref, className: ["table-cell", className].join(" "), ...props }));
TableCell.displayName = "TableCell";
const TableCaption = React.forwardRef(({ className, ...props }, ref) => /* @__PURE__ */ React.createElement(
  "caption",
  {
    ref,
    className: ["table-caption", className].join(" "),
    ...props
  }
));
TableCaption.displayName = "TableCaption";

export { Table, TableBody, TableCaption, TableCell, TableFooter, TableHead, TableHeader, TableRow };
